aptos-move/framework/aptos-framework/sources/voting.spec.move

```diff
@@ -169,11 +183,29 @@ spec aptos_framework::voting {
         proposal_id: u64,
     ): u64 {
         use aptos_framework::chain_status;
-        requires chain_status::is_operating(); // Ensures existence of Timestamp
-        // Addition of yes_votes and no_votes might overflow.
         pragma addition_overflow_unchecked;
+
+        requires chain_status::is_operating(); // Ensures existence of Timestamp
+
         include AbortsIfNotContainProposalID<ProposalType>;
-        // Any way to specify the result?
+
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals, proposal_id);
+        let early_resolution_threshold = option::spec_borrow(proposal.early_resolution_vote_threshold);
+        let voting_period_over = timestamp::now_seconds() > proposal.expiration_secs;
+        let be_resolved_early = option::spec_is_some(proposal.early_resolution_vote_threshold) &&
+                                    (proposal.yes_votes >= early_resolution_threshold ||
+                                     proposal.no_votes >= early_resolution_threshold);
+        let voting_closed = voting_period_over || be_resolved_early;
+        /// Succeeded or Failed
+        ensures voting_closed ==> if (proposal.yes_votes > proposal.no_votes && proposal.yes_votes + proposal.no_votes >= proposal.min_vote_threshold) {
+            result == PROPOSAL_STATE_SUCCEEDED
+        } else {
+            result == PROPOSAL_STATE_FAILED
+        };
+
+        /// Pending
```

Comment:
Can you use `//` instead of `///`? Could you give a bit more descriptive comment?